src/components/toolbar/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import { translate } from 'react-i18next';
+import Search from '../search'
+import style from './style.styl';
+
+const Styles = (a) => a.filter(b => !!b).join(' ');
+
+class Button extends Component {
+    constructor (props) {
+        super()
+        this.state = {
+            active: props.active || false,
+            update: props.update || false
+        }
+    }
+
+    onUpdate() {
+        this.setState((prev) => ({update: true}));
+    }
+
+    onClick() {
+        let {props, state} = this
+        props.toogle && this.setState((prev) => ({active: !!!prev.active}));
+        props.action.bind(this)
+    }
+
+    render () {
+        let {props, state} = this
+        let classList = Styles([style.button, state.active && style.active, state.update && style.update])
```

Comment:
`Styles()` is for filter / add multiple classes, not sure how to validate component `state`